cEP-0016.md

```diff
@@ -0,0 +1,382 @@
+and tries to merge master onto the pull request. Rebase and merge on GitHub will
+always update the committer information and create new commit SHAs which is not
+correct. These commits generate a default commit message that is very concise
+and not according to coala’s standards. The coala project wants to allow
+`git revert` commits, but does not want to accept `git merge` commits as the
+repositories are strictly fast-forward merges that do not have an extra commit.
+
+In cases where some tests are not required by any pull request made to coala,
+some special sequences can be added to git commit messages to instruct tools to
+operate in a certain way.
+
+This project is about detecting such commits and verifying that they are correct
+according to the configuration settings in `.coafile`
+
+## Background
+
+`.coafile` is a project-wide coala configuration file which is used to specify
+what is checked with which bears for a project.
+
+The workaround that coala uses currently to handle the problem of Travis merge
+commits in `travis.yml` is:
+
+```
+if [[ "$TRAVIS_PULL_REQUEST" != "false" ]]; then
+  sed -e '/bears = GitCommitBear/d' .coafile > .coafile.new
+  mv .coafile.new .coafile
+fi
+```
+
+Below are some code snippets of `.coafile` from different projects depicting
+their use of `GitCommitBear`.
+
+Projects- [coala/coala](https://github.com/coala/coala), [coala/projects](https://github.com/coala/projects), [coala/documentation](https://github.com/coala/documentation)
+
+```
+[commits]
+bears = GitCommitBear
+shortlog_regex = ([^:]*|[^:]+: [A-Z0-9*].*)
+shortlog_trailing_period = False
+```
+
+Projects- [coala/coala-bears](https://github.com/coala/coala-bears), [coala/coala-quickstart](https://github.com/coala/coala-quickstart), [coala/corobo](https://github.com/coala/corobo)
+
+```
+[all.commit]
+bears = GitCommitBear
+shortlog_trailing_period = False
+shortlog_regex = ([^:]*|[^:]+: [A-Z0-9*].*)
+body_close_issue = True
+body_close_issue_full_url = True
+body_close_issue_on_last_line = True
+body_enforce_issue_reference = False
+```
+
+Project- [coala/coala-atom](https://github.com/coala/coala-atom)
+
+```
+[commit]
+bears = GitCommitBear
+```
+
+## Proposed Change
+
+This project would allow organizations to customize the `.coafile`
+to allow following settings for `GitCommitBear`.
+
+```
+[commit]
+bears = GitCommitBear
+allow_skip_ci_build = False
+allow_git_merge_commit = False
+allow_git_revert_commit = True
+```
+
+Travis merge commits, `git revert` and `git merge` commits would be inspected
+by making changes in the `GitCommitBear`. For commits that disable CI build,
+a new meta bear would be implemented.
+
+Here is the detailed implementation stepwise:
+
+1. We start by identifying different types of special commits namely Travis
+   merge commits, `git revert` , `git merge` and commits that disable CI builds.
+2. Each type of commit is then handled as a separate case.
+3. For Travis merge commits:
```

Comment:
`GitHub Pull Request temporary merge commits` is fine for me..

Since it's really "temporary", it seems like GitHub removes this `merge` reference when the pull request is "Merged" (not closed).
When I did `git ls-remote upstream` (`upstream` is my git remote which points to `https://github.com/coala/coala-bears`), I got:
```
... (too many results)
6da7f70cd07149b10f580242a435ac09189b6988	refs/pull/257/head
5421d76c361e71cb94373fba1197db767740c6fe	refs/pull/258/head
0baf3e0925677c2537351f0a4d88e9c5568f238c	refs/pull/262/head
24a3b3807fd7ac0fd0ab0332858ac38fea2d63ef	refs/pull/263/head
2d2a23f9c30dffd72e1c23202712f670410180f7	refs/pull/263/merge
....
```
If we open coala/coala-bears#263, it's already closed but the `merge` reference is still there, and the others without the `merge` reference such as coala/coala-bears#262 mean that the pull request has been merged

And yeah, it turns out this "merge" commit are made by GitHub and not by Travis..

PS from https://blog.jetbrains.com/teamcity/2013/02/automatically-building-pull-requests-from-github-with-teamcity/ :

> When a pull request is made, GitHub automatically creates a reference that holds the pull request as well as one that is a merge with the master branch.

Reply:
I'm fine with both names, as long as it's explained in the doc with a link to what they are (so it's clear for whoever read this that it's a travis weirdness) 😉